spec.md

```diff
@@ -547,212 +547,129 @@ See [@sec:mem] for more details.
-The following example specifies a twenty element vector, each of which is a ten element vector of 16-bit unsigned integers.
+TODO: *Any* type? What about probes and properties?
 
-``` firrtl
+```firrtl
 UInt<16>[10][20]
 ```
 
+TODO: In analogy to 0-width integers, write a section on 0-length vecs.
+
 ### Bundle Types
 
-A bundle type is used to express a collection of nested and named types.
-All fields in a bundle type must have a given name, and type.
-All names must be legal identifiers.
+A bundle type is used to represent a collection of values.
+They can also be used to facilitate bidirectional connections between circuit components.
+
+A bundle type consists of a set of fields.
+Each field has a name and a type.
+Fields may also be flipped.
 
 The following is an example of a possible type for representing a complex number.
 It has two fields, `real`{.firrtl}, and `imag`{.firrtl}, both 10-bit signed integers.
 
-``` firrtl
-{real: SInt<10>, imag: SInt<10>}
+```firrtl
+{ real: SInt<10>, imag: SInt<10> }
 ```
 
-Additionally, a field may optionally be declared with a *flipped* orientation.
+Here is an example of a bundle with a flipped field.
+Because the `ready` field is marked with the keyword `flip`, it will indicate the flow will be opposite of the `word` and `valid` fields.
 
-``` firrtl
-{word: UInt<32>, valid: UInt<1>, flip ready: UInt<1>}
+```firrtl
+{ word: UInt<32>, valid: UInt<1>, flip ready: UInt<1> }
 ```
 
-In a connection between circuit components with bundle types, the data carried by the flipped fields flow in the opposite direction as the data carried by the non-flipped fields.
+As an example of how `flip` works in context, consider a module declared like this:
 
-As an example, consider a module output port declared with the following type:
-
-``` firrtl
-output a: {word: UInt<32>, valid: UInt<1>, flip ready: UInt<1>}
+```firrtl
+module Processor :
+  input enq: { word: UInt<32>, valid: UInt<1>, flip ready: UInt<1> }
+  ; ...
 ```
 
-In a connection to the `a`{.firrtl} port, the data carried by the `word`{.firrtl} and `valid`{.firrtl} sub-fields will flow out of the module, while data carried by the `ready`{.firrtl} sub-field will flow into the module.
-More details about how the bundle field orientation affects connections are explained in [@sec:connects].
+This defines a module `Processor` with a single input port `enq` which enqueues data using a ready-valid interface.
+Because `enq` is a single port, we can connect to it with a single `connect` statement (see [@sec:connects]).
+In the final hardware, however, while the `word` and `valid` fields point *into* the module, this port has a flipped field `ready` which points *out of* the module instead.
 
-As in the case of vector types, a bundle field may be declared with any type, including other aggregate types.
+The types of each field may be any type, including other aggregate types.
```

Comment:
This is a bit of a hard transition from the previous paragraph. I thought this was continuing the example above.

Perhaps this would be better located at the start of the sub-section.